### PR TITLE
Packaging markdown files into jar. Structure for markdown files within adapter is changed.

### PR DESCRIPTION
**Pull request by nikolag-ikor**
# Description

Added markdown files in jar. When running the adapter with jar, markdown files from adapter application module should be included and shown under /actuator/info endpoint.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Testing steps:
1. Create new adapter from archetype.
2. do mvn clean install in your adapter
3. Run the adapter and check if everything is there under /actuator/info
4. Run the jar of adapter and check actuator/info

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing (regression) unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sip-archetype/src/main/resources/archetype-resources/README.md [hunk 1]
```diff
@@ -8,9 +8,10 @@ This is an auto-generated project and you can find more information about develo
...
 
-Please use this document to describe how to build and run your adapter.
+Please use `readme.md` document to describe how to build and run your adapter. This document can be found in your
+adapter application module under `main/resources/documents`.
 
 We encourage you to explore our features and hope you enjoy developing your adapter with SIP.
 
 We would also like to have you in our team by contributing to the framework. Please check here how to contribute:
-https://git.ikor.cloud/sip-2-product/sip/sip-framework/-/blob/develop/CONTRIBUTING.md
+https://git.ikor.cloud/sip-2-product/sip/sip-framework/-/blob/develop/CONTRIBUTING.md
```
Comment (LetoBukarica) [inline sip-archetype/src/main/resources/archetype-resources/README.md]: I think this should go to:
https://github.com/IKOR-GmbH/sip-framework/blob/develop/docs/CONTRIBUTING.md.

There is an another reference in the file that should go to README.md, not included in this changelog.

File: sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java [hunk 2]
```diff
@@ -2,28 +2,17 @@
...
-import java.io.FileWriter;
-import java.io.IOException;
 import java.util.LinkedHashMap;
 import java.util.List;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.slf4j.LoggerFactory;
 import org.springframework.boot.actuate.info.Info;
 
 class MarkdownFilesContributorTest {
```
Comment (LetoBukarica) [inline sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java]: Can we have a little bit more tests to see how this works. What if there are 2 documents in the /documents folder, any error conditions?

File: sip-core/src/test/resources/documents/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Welcome!
+
+**Thank you for using SIP Framework!**
+
+You have successfully created ${artifactId}. We recommend you to describe the adapter you are developing within this file.
+
+This is an auto-generated project and you can find more information about developing approach, project structure and
+sip framework concept at:
+https://git.ikor.cloud/sip-2-product/sip/sip-framework/-/blob/develop/README.md
```
Comment (LetoBukarica) [inline sip-core/src/test/resources/documents/README.md]: Same as in previous file, references are to GitLab

**Review by LetoBukarica — CHANGES_REQUESTED** [review]
Looks good, small tweaks needed.

File: sip-archetype/src/main/resources/archetype-resources/README.md (hunk 1 above)
Comment (nikolag-ikor) [reply sip-archetype/src/main/resources/archetype-resources/README.md]: You are totally right. I will change these links, thanks!

File: sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java (hunk 2 above)
Comment (nikolag-ikor) [reply sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java]: As a result of our short discussion, I will try to implement it without using static files and will try to do it by creating files dynamically.

**Review by Dzuci — DISMISSED** [review]
Tested with OK behavior.

File: sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java (hunk 2 above)
Comment (nikolag-ikor) [reply sip-core/src/test/java/de/ikor/sip/foundation/core/actuator/routes/MarkdownFilesContributorTest.java]: Refactored as desired. Reviewed by pair programming with @LetoBukarica.

**Review by LetoBukarica — APPROVED** [review]

**Review by Dzuci — APPROVED** [review]
Tested with OK behavior.

**Pull request merged**